controllers/discovery/searchDomain.js

```diff
@@ -12,11 +12,6 @@ const {CREDDER_MIN_SCORE} = require('../../helpers/constants');
 const SearchDomain = async (req, res) => {
   const {q} = req.query;
   const {userId} = req;
-  if (!q || q?.length < 2)
-    return res.status(200).json({
-      success: true,
-      message: 'Your search characters is too few, please input 3 or more characters for search'
-    });
 
   try {
     const domains = await sequelize.query(
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Removing the character length check for the search query might lead to performance issues. If a user enters a single character or an empty string, it could potentially return a large number of results, which could slow down the system. Consider adding a check to prevent this scenario.
```diff
+  if (!q || q?.length < 1)
+    return res.status(200).json({
+      success: true,
+      message: 'Your search characters is too few, please input 2 or more characters for search'
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->